lib/membrane_http_adaptive_stream/storage.ex

```diff
@@ -7,7 +7,10 @@ defmodule Membrane.HTTPAdaptiveStream.Storage do
 
   @type config_t :: struct
   @type state_t :: any
-  @type callback_result_t :: :ok | {:error, reason :: any}
+
+  @type stateless_result_t :: :ok | {:error, reason :: any}
+  @type stateful_result_t :: {stateless_result_t, state_t}
```

Comment:
This is used only in the line below. Shouldn't it be used everywhere, instead of stateless_result_t?

Reply:
`stateful_result_t` has `state_t` (which is `any`) as the second element. We have a better type to describe `Storage` structs which is `t`, so I used `{stateless_result_t, t}` instead. Technically, we could use `stateful_result_t` everywhere and dialyzer would not complain, but I wanted to be more precise with the typing.